step-plans/step-plans-core/src/main/java/step/core/plans/builder/PlanBuilder.java

```diff
@@ -103,4 +107,29 @@ private void addToCurrentParent(AbstractArtefact artefact) {
 		AbstractArtefact parent = stack.peek();
 		parent.addChild(artefact);
 	}
+
+
+	public PlanBuilder withBefore(AbstractArtefact... before) {
+		ChildrenBlock childrenBlock = new ChildrenBlock();
+		childrenBlock.setSteps(List.of(before));
+		applyToLastNode((a) -> a.setBefore(childrenBlock));
+		return this;
+	}
+
+	public PlanBuilder withAfter(AbstractArtefact... after) {
+		ChildrenBlock childrenBlock = new ChildrenBlock();
+		childrenBlock.setSteps(List.of(after));
+		applyToLastNode((a) -> a.setAfter(childrenBlock));
+		return this;
+	}
+
+	public PlanBuilder withSkip() {
+		return applyToLastNode(a -> a.setSkipNode(new DynamicValue<>(true)));
+	}
+
+	private PlanBuilder applyToLastNode(Consumer<AbstractArtefact> c) {

```

Comment:
Shouldn't this be called applyToParentNode?

Reply:
Refactored, being a stack I guess it's more clear